android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java

```diff
@@ -504,6 +515,33 @@ public void propertyChanged(String key, Object oldValue, Object newValue, KrollP
 		}
 	}
 
+	private void zoomBy(WebView webView, float scale)
+	{
+		if (webView != null) {
+			if (scale < 0.0f) {
+				scale = 0.0f;
+			} else if (scale >= 100.0f) {
+				scale = 100.0f;
+			}
+			webView.zoomBy(scale);
+		}
+	}
+
+	public void zoomBy(float scale)
+	{
+		zoomBy(getWebView(), scale);
+	}
+
+	public float getZoomLevel()
+	{
+		return zoomLevel;
+	}
+
+	public void setZoomLevel(float value)
+	{
+		zoomLevel = value;
+	}
+
```

Comment:
Having a public `setZoomLevel()` method that doesn't actually change the WebView's zoom level is a confusing API. I understand why you're doing this, but there is a better way.

I recommend that you remove the `setZoomLevel()` method and the `zoomLevel` member variable from the `TiUIWebView` class. You should instead add a `zoomLevel` member variable and `getZoomLevel()` method to the `TiWebViewClient` class and update the member variable there via its onScaleChanged() listener. Your `TiUIWebView.getZoomLevel()` method can then simply return the client's zoom level like this...

```java
public float getZoomLevel()
{
	return (this.client != null) ? this.client.getZoomLevel() : 1.0f;
}
```